extensions/emmet/src/abbreviationActions.ts

```diff
@@ -435,17 +442,32 @@ function expandAbbr(input: ExpandAbbreviationInput): string | undefined {
 
 	try {
 		// Expand the abbreviation
-		let expandedText = helper.expandAbbreviation(input.abbreviation, expandOptions);
+		let expandedText;
 
 		if (input.textToWrap) {
+			let parsedAbbr = helper.parseAbbreviation(input.abbreviation, expandOptions);
+			if (input.rangeToReplace.isSingleLine) {
+
+				// Fetch innermost element in the expanded abbreviation
+				let lastNode = parsedAbbr;
+				let tagName = parsedAbbr.name || '';
+				while (lastNode.children && lastNode.children.length > 0) {
+					lastNode = lastNode.children[lastNode.children.length - 1];
+					tagName = lastNode.name;
+				}
+
+				// If wrapping with a block element, insert newline and expand again
```

Comment:
These comments need to be updated as we are not expanding "again" anymore

Reply:
You're right